rotkehlchen/tests/db/test_db.py

```diff
@@ -1274,3 +1288,177 @@ def test_int_overflow_at_tuple_insertion(database, caplog):
+
+
+def test_all_trade_types_in_db(database):
+    """
+    Test that all trade_types in DB deserialize to a valid TradeType
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    trade_types = cursor.execute("SELECT type, seq from trade_type")
+
+    # We deserialize, then serialize and compare the result
+    for trade_type, seq in trade_types:
+        deserialized_trade_type = deserialize_trade_type_from_db(trade_type)
+        assert deserialized_trade_type.value == seq
+        assert TradeType(seq).serialize_for_db() == trade_type
+        trade_type_name = deserialize_trade_type(str(deserialized_trade_type))
+        assert trade_type_name == deserialized_trade_type
+
+
+def test_all_trade_types_are_present(database):
+    """
+    Test that all TradeType are present in db
+    """
+    cursor = database.conn.cursor()
+    for trade_type in TradeType:
+        r = cursor.execute(
+            "SELECT EXISTS(SELECT 1 FROM trade_type WHERE seq=?)",
+            (trade_type.value,))
+        assert r.fetchone() == (1,)
+
+
+def test_all_action_types_in_db(database):
+    """
+    Test that all action_type in DB deserialize to a valid ActionType
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    action_types = cursor.execute("SELECT type, seq from action_type")
+
+    # We deserialize, then serialize and compare the result
+    for action_type, seq in action_types:
+        deserialized_action_type = deserialize_action_type_from_db(action_type)
+        assert deserialized_action_type.value == seq - 1
+        action_type_name = deserialize_action_type(str(deserialized_action_type))
+        assert action_type_name == deserialized_action_type
+
+
+def test_all_action_types_are_present(database):
+    """
+    Test that all ActionType are present in db
+    """
+    cursor = database.conn.cursor()
+    for action_type in ActionType:
+        r = cursor.execute(
+            "SELECT EXISTS(SELECT 1 FROM action_type WHERE seq=?)",
+            (action_type.value + 1,))
+        assert r.fetchone() == (1,)
+
+
+def test_all_ledger_action_types_in_db(database):
+    """
+    Test that all ledger_action_type in DB deserialize to a valid LedgerActionType
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    ledger_action_types = cursor.execute("SELECT type, seq from ledger_action_type")
+
+    # We deserialize, then serialize and compare the result
+    for ledger_action_type, seq in ledger_action_types:
+        deserialized_ledger_action_type = deserialize_ledger_action_type_from_db(
+            ledger_action_type,
+        )
+        assert deserialized_ledger_action_type.value == seq - 1
+        ledger_action_type_name = deserialize_ledger_action_type(
+            str(deserialized_ledger_action_type),
+        )
+        assert ledger_action_type_name == deserialized_ledger_action_type
+
+
+def test_all_ledger_action_types_are_present(database):
```

Comment:
We have some similar tests for ledger actions already here: https://github.com/rotki/rotki/blob/021334a1fd7a4e5eb3c90736b3508c56878efd45/rotkehlchen/tests/unit/test_ledger_actions.py#L21-L39